polkadot/runtime/parachains/src/assigner_on_demand/mod.rs

```diff
@@ -599,6 +605,8 @@ where
 				// Only update storage on change
 				if new_traffic != old_traffic {
 					queue_status.traffic = new_traffic;
+
+					// emit the event for updating new price
```

Comment:
See main body of the review for pointers but this event needs changes

Reply:
Thank you very much for your willingness to help and make me understand the requirement of the issue. 

I have implemented the changes as you explained and pushed another commit to cater for them.